src/Features/CSharp/Portable/DocumentSymbols/CSharpDocumentSymbolsService.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.DocumentSymbols;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Tags;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.DocumentSymbols
+{
+    [ExportLanguageService(typeof(IDocumentSymbolsService), LanguageNames.CSharp), Shared]
+    internal class CSharpDocumentSymbolsService : AbstractDocumentSymbolsService
+    {
+        private static readonly SymbolDisplayFormat s_typeFormat =
+            SymbolDisplayFormat.CSharpErrorMessageFormat.AddGenericsOptions(SymbolDisplayGenericsOptions.IncludeVariance);
+
+        private static readonly SymbolDisplayFormat s_memberFormat =
+            new(
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters |
+                               SymbolDisplayMemberOptions.IncludeExplicitInterface,
+                typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameOnly,
+                parameterOptions: SymbolDisplayParameterOptions.IncludeType |
+                                  SymbolDisplayParameterOptions.IncludeName |
+                                  SymbolDisplayParameterOptions.IncludeDefaultValue |
+                                  SymbolDisplayParameterOptions.IncludeParamsRefOut,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes |
+                                      SymbolDisplayMiscellaneousOptions.AllowDefaultLiteral |
+                                      SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+        private static readonly ImmutableArray<string> s_constantTag = ImmutableArray.Create(WellKnownTags.Constant);
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpDocumentSymbolsService()
+        {
+        }
+
+        protected override ISymbol? GetSymbol(SemanticModel semanticModel, SyntaxNode node, CancellationToken cancellationToken)
+            => node switch
+            {
+                NamespaceDeclarationSyntax or GlobalStatementSyntax => null,
+                MemberDeclarationSyntax member => semanticModel.GetDeclaredSymbol(member, cancellationToken),
+                LabeledStatementSyntax labeledStatement => semanticModel.GetDeclaredSymbol(labeledStatement, cancellationToken),
```

Comment:
this is interesting.  do we put labels into our nav bars??

Reply:
No. We only put types and their top-level members into the navbars.